src/Platform/Microsoft.Testing.Platform/DotnetTest/IPC/Serializers/DiscoveredTestMessageSerializer.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.IPC.Models;
+
+namespace Microsoft.Testing.Platform.IPC.Serializers;
+
+/*
+    |---FieldCount---| 2 bytes
+
+    |---File Uid Id---| (2 bytes)
+    |---File Uid Size---| (4 bytes)
+    |---File Uid Value---| (n bytes)
+
+    |---File DisplayName Id---| (2 bytes)
+    |---File DisplayName Size---| (4 bytes)
+    |---File DisplayName Value---| (n bytes)
+
+    |---File ExecutionId Id---| (2 bytes)
+    |---File ExecutionId Size---| (4 bytes)
+    |---File ExecutionId Value---| (n bytes)
+*/
+
+internal sealed class DiscoveredTestMessageSerializer : BaseSerializer, INamedPipeSerializer
+{
+    public int Id => 10;
```

Comment:
It could be interesting move the `MessageId` as a prop together with the constants to have all numbers in one place. Followup maybe.

Reply:
done.